edge-apps/calendar/src/events.ts

```diff
@@ -43,6 +44,73 @@ const getDateRangeForViewMode = (viewMode: ViewMode) => {
   return { startDate, endDate }
 }
 
+export const fetchCalendarEventsFromAPI = async (): Promise<
+  CalendarEvent[]
+> => {
+  try {
+    const { calendar_mode: viewMode } = screenly.settings
+    const { startDate, endDate } = getDateRangeForViewMode(viewMode as ViewMode)
+
+    console.log('Fetching access token...')
+    // Get access token with retry logic
+    const accessToken = await retryWithBackoff(() => getAccessToken(), 3, 1000)
+    console.log('Access token retrieved:', accessToken ? 'Success' : 'Failed')
+
+    // Fetch events from Google Calendar API
+    const calendarId = 'primary'
+    const timeMin = startDate.toISOString()
+    const timeMax = endDate.toISOString()
+    const apiUrl = `https://www.googleapis.com/calendar/v3/calendars/${calendarId}/events?timeMin=${encodeURIComponent(timeMin)}&timeMax=${encodeURIComponent(timeMax)}&singleEvents=true&orderBy=startTime`
+
+    console.log('Fetching calendar events from Google Calendar API...')
+    console.log('API URL:', apiUrl)
+    console.log('Time range:', { timeMin, timeMax })
+
+    const response = await fetch(apiUrl, {
+      headers: {
+        Authorization: `Bearer ${accessToken}`,
+      },
+    })
+
+    console.log('Google Calendar API response status:', response.status)
+
+    if (!response.ok) {
+      throw new Error(
+        'Failed to fetch calendar events from Google Calendar API',
+      )
+    }
+
+    const data = await response.json()
+    console.log(
+      'Calendar events data received:',
+      data.items ? `${data.items.length} events` : 'No events',
+    )
+
+    const events: CalendarEvent[] = []
+
+    if (data.items && Array.isArray(data.items)) {
```

Comment:
Can it be a non-array?

Also, always try to use early return rule
```suggestion
    if (!data.items) {
        return;
    }
```